Sources/SwiftDiscord/DiscordEndpoint.swift

```diff
@@ -17,6 +17,49 @@
 
 import Foundation
 
+/**
+ * An HTTP content-type.  Common options are available as enum values, but if you need something special, use .other("My-Special-Type")
+ */
+public enum HTTPContentType: CustomStringConvertible {
+    case json
+    case other(String)
+
+    public var description: String {
+        switch self {
+        case .json:
+            return "application/json"
+        case let .other(type):
+            return type
+        }
+    }
+}
+
+/**
+ * An HTTP Request Method.  This includes any associated data.
+ */
+public enum HTTPMethod {
```

Comment:
Do you have any thoughts on the name?  I don't want to call it a request because its data gets added to an existing request object, and running `createRequest` and giving it a request would be confusing.